plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2996,6 +3021,37 @@ public boolean delegateMaterializedViewRefreshToConnector(ConnectorSession sessi
         return hiveMaterializedViewMetadata.refreshMaterializedView(session, name);
     }
 
+    @Override
+    public Optional<CatalogSchemaTableName> redirectTable(ConnectorSession session, SchemaTableName tableName)
```

Comment:
Can you please update this to https://github.com/trinodb/trino/pull/9870?

Reply:
> Can you please update this to #9870?

I've taken liberty to update the code myself. 
I also copied & adapted an awesome test template done by @ssheikin too.
posted https://github.com/trinodb/trino/pull/10173 with the changes

Reply:
This credits belong to @MiguelWeezardo

Reply:
> These credits belong to @MiguelWeezardo

thanks @ssheikin for the comment. added @MiguelWeezardo as co-author in the other PR